src/many/src/types/identity/cose.rs

```diff
@@ -56,24 +59,45 @@ impl CoseKeyIdentity {
         Self {
             identity: Identity::anonymous(),
             key: None,
+            hsm: false,
         }
     }
 
-    pub fn from_key(key: CoseKey) -> Result<Self, String> {
+    pub fn from_key(key: CoseKey, hsm: bool) -> Result<Self, String> {
```

Comment:
Can we make this private? I don't think people should use this by itself (without using `from_hsm` or `from_pem`).

Reply:
We're using it in `src/many/src/message.rs`. Would a `pub(crate)` be good enough for you until we refactor?

Reply:
Yes.